lib/tubby.rb

```diff
@@ -55,6 +55,18 @@ def doctype!
     end
 
     def __attrs!(attrs)
+      if attrs.key?(:data) && attrs[:data].is_a?(Hash)
+        # Flatten present `data: {k1: v1, k2: v2}` attribute, inserting `data-k1: v1, data-k2: v2`
+        # into exact place where the attribute was
+        attrs = Hash[attrs.map do |key, value|
+          if key == :data && value.is_a?(Hash)
+            value.map { |k, v| [:"data-#{k}", v] }
+          else
+            [[key, value]]
+          end
+        end.flatten(1)]
```

Comment:
This is a great case for `flat_map`!

```suggestion
        attrs = Hash[attrs.flat_map do |key, value|
          if key == :data && value.is_a?(Hash)
            value.map { |k, v| [:"data-#{k}", v] }
          else
            [key, value]
          end
        end]
```

Reply:
:facepalm: Somehow assumed without checking `flat_map` was not there in 2.0. But it was.